flink-table/flink-table-runtime/src/main/java/org/apache/flink/table/runtime/functions/scalar/ArrayJoinFunction.java

```diff
@@ -0,0 +1,78 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.table.runtime.functions.scalar;
+
+import org.apache.flink.annotation.Internal;
+import org.apache.flink.table.data.ArrayData;
+import org.apache.flink.table.data.StringData;
+import org.apache.flink.table.functions.BuiltInFunctionDefinitions;
+import org.apache.flink.table.functions.SpecializedFunction;
+import org.apache.flink.table.types.CollectionDataType;
+import org.apache.flink.table.types.DataType;
+import org.apache.flink.util.FlinkRuntimeException;
+
+import javax.annotation.Nullable;
+
+/** Implementation of {@link BuiltInFunctionDefinitions#ARRAY_JOIN}. */
+@Internal
+public class ArrayJoinFunction extends BuiltInScalarFunction {
+
+    private final ArrayData.ElementGetter elementGetter;
+
+    public ArrayJoinFunction(SpecializedFunction.SpecializedContext context) {
+        super(BuiltInFunctionDefinitions.ARRAY_JOIN, context);
+        final DataType elementDataType =
+                ((CollectionDataType) context.getCallContext().getArgumentDataTypes().get(0))
+                        .getElementDataType();
+        elementGetter = ArrayData.createElementGetter(elementDataType.getLogicalType());
+    }
+
+    public @Nullable StringData eval(
+            ArrayData array, StringData delimiter, StringData... nullReplacement) {
+        try {
+            if (array == null
+                    || delimiter == null
+                    || (nullReplacement.length != 0 && nullReplacement[0] == null)) {
```

Comment:
could you add a test for that?
array != null && delimiter != null && nullReplacement[] == null

Reply:
```
.testResult(
                                call("ARRAY_JOIN", $("f0"), null),
                                "ARRAY_JOIN(f0, null)",
                                null,
                                DataTypes.STRING())
```
this one

Reply:
https://github.com/apache/flink/pull/22785/files#r1244294436